### PR TITLE
Add missing docstrings for miscellaenous functions and classes

### PR DESCRIPTION
**Pull request by dstrain115**
- These classes and functions are exposed in the global name space
but are missing docstrings.
- Added short docstrings for each object.

### CODE REVIEW

File: cirq-core/cirq/circuits/circuit.py [hunk 1]
```diff
@@ -92,6 +92,8 @@ def foo(circuit: CIRCUIT_TYPE) -> CIRCUIT_TYPE:
 
 
 class Alignment(enum.Enum):
+    """Alignment Option for combining/zipping two circuits together."""
```
Comment (dabacon) [inline cirq-core/cirq/circuits/circuit.py:95]: nit: reason to capitalize Option?

File: cirq-core/cirq/protocols/json_serialization.py [hunk 2]
```diff
@@ -819,6 +838,28 @@ def read_json_gzip(
     gzip_raw: Optional[bytes] = None,
     resolvers: Optional[Sequence[JsonResolver]] = None,
 ):
+    """Read a gzipped JSON file that optionally contains cirq objects.
+
+    Args:
+        file_or_fn: A filename (if a string or `pathlib.Path`) to read from, or
+            an IO object (such as a file or buffer) to read from, or `None` to
+            indicate that `json_text` argument should be used. Defaults to
```
Comment (dabacon) [inline cirq-core/cirq/protocols/json_serialization.py:846]: think this should be `gzip_raw` and not `json_text`

**Review by dabacon — CHANGES_REQUESTED** [review]
forgot to mark requested changes

File: cirq-core/cirq/circuits/circuit.py (hunk 1 above)
Comment (dstrain115) [reply cirq-core/cirq/circuits/circuit.py]: Done

File: cirq-core/cirq/protocols/json_serialization.py (hunk 2 above)
Comment (dstrain115) [reply cirq-core/cirq/protocols/json_serialization.py]: Done.

File: cirq-core/cirq/sim/state_vector_simulator.py [hunk 3]
```diff
@@ -110,6 +110,10 @@ class StateVectorStepResult(
 @deprecated_class(deadline='v0.16', fix='This class is no longer used.')
 @value.value_equality(unhashable=True)
 class StateVectorSimulatorState:
+    """Object representing current state of the simulator.
+    Includes the state vector, qubit map, and shape information.
```
[the remark this answers is not in the document]
Comment (dstrain115) [reply cirq-core/cirq/sim/state_vector_simulator.py:115]: Done.

File: cirq-core/cirq/value/classical_data.py [hunk 4]
```diff
@@ -23,6 +23,8 @@
 
 
 class MeasurementType(enum.IntEnum):
+    """Type of a measurement, whether a measurement or channel."""
```
[the remark this answers is not in the document]
Comment (dstrain115) [reply cirq-core/cirq/value/classical_data.py]: Added more information.  I think what I wrote is correct.  Maybe @95-martin-orion can verify.

[the remark this answers is not in the document]
Comment (95-martin-orion) [reply cirq-core/cirq/value/classical_data.py]: This is related to the `_records` and `_channel_records` fields of the classical data store object [here](https://github.com/quantumlib/Cirq/blob/b1a5a39af9c3045221290784b1dcf04b53ba7ad0/cirq-core/cirq/value/classical_data.py#L134-L136). In short, `MEASUREMENT` indicates the data is in `_records` and includes per-qubit results, while `CHANNEL` indicates the data is in `_channel_records` and has a single "which-operator" value.

**Review by dabacon — APPROVED** [review]
LGTM

**Pull request merged**